std/range/primitives.d

```diff
@@ -161,17 +161,11 @@ Params:
 Returns:
     true if R is an InputRange, false if not
  */
-template isInputRange(R)
-{
-    enum bool isInputRange = is(typeof(
-    (inout int = 0)
-    {
-        R r = R.init;     // can define a range object
-        if (r.empty) {}   // can test for empty
-        r.popFront;       // can invoke popFront()
-        auto h = r.front; // can get the front of the range
-    }));
-}
+enum bool isInputRange(R) =
+    is(typeof((R r) => R.init))
```

Comment:
The previous constraint verified that this works: `R r = R.init;`. That passes even for ranges with both this() and this(this) disabled, so you're right about that.

The only types rejected by this would be those that play shenanigans with `init`, i.e. have it return a different type etc. We don't want to support such styles. I'll remove the line.